Product class

eid-sonkrot opened the pull request.

Commit:
Implement Product Class Modification
In this commit, we made significant enhancements to the Product class, adding three essential properties: Name, Price, and Quantity. These properties are now publicly accessible through setter and getter methods, allowing smooth interaction with the Product class.

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Text;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using System.Xml.Linq;
+using ConsoleTables;
```
please remove the unused imports

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
 {
     internal class Product
     {
+        public string Name { get; set; }
+        public double Price { get; set; }
+        public int Quantity { get; set; }
+        public Product()
```
```suggestion
        public string Name { get; set; }
        public double Price { get; set; }
        public int Quantity { get; set; }

        public Product()
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+        public string Name { get; set; }
+        public double Price { get; set; }
+        public int Quantity { get; set; }
+        public Product()
+        {
+            this.Name = null;
+            this.Price = 0;
+            this.Quantity = 0;
+        }
```
please remove this constructor, since the default value of string is null, double is 0, and for int is 0.

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+           
+            do
+            {
+                string Name;
```
```suggestion
                var Name = (string?) null;
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                string Name;
+                // Inform the user to enter the product's name
+                Console.Write("Enter the product's name: ");
+                Name = Console.ReadLine();
+                if(Regex.IsMatch(Name, @"^[A-Za-z]+$"))
```
```suggestion
                Name = Console.ReadLine();

                if (Regex.IsMatch(Name, @"^[A-Za-z]+$"))
```

Alatrash814 replied:
please move the Regex value to a helper, because it will be used in many places.

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+        
+        public void Input()
+        {   //Validating the input to ensure its correctness
+            bool IsValid = false;
```
```suggestion
            var isValid = false;
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                {
+                    Console.WriteLine("Invalid input. Please enter a Name contain Alpapitic character");
+                }
+            } while (!IsValid);
+            this.Name = Name;
```
```suggestion
            } while (!IsValid);

            this.Name = Name;
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                }
+            } while (!IsValid);
+            this.Name = Name;
+            IsValid = false;
+            do
```
```suggestion
            IsValid = false;

            do
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+            IsValid = false;
+            do
+            {
+                Double Price;
```
```suggestion
                var price = (Double?) null;
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                // Inform the user to enter the product's price
+                Console.Write("Enter the product's price: ");
+
+                if (double.TryParse(Console.ReadLine(), out Price))
```
```suggestion
                var isValidPrice = double.TryParse(Console.ReadLine(), out Price)

                if (isValidPrice)
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                else
+                {
+                    Console.WriteLine("Invalid input. Please enter a name containing only alphabetical characters.");
+                    
```
please remove this empty line.

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                }
+            } while (!IsValid);
+
+
```
please remove this empty line.

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+            IsValid = false;
+            do
+            {
+                int Quantity;
```
```suggestion
                var quantity = (int?) null;
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+                // Inform the user to enter the product's quantity
+                Console.Write("Enter the product's quantity: ");
+
+                if (int.TryParse(Console.ReadLine(), out Quantity))
```
```suggestion
                var isValidQuantity = int.TryParse(Console.ReadLine(), out Quantity);

                if (isValidQuantity )
```

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -1,12 +1,88 @@
+            this.Quantity = 0;
+        }
+        
+        public void Input()
```
please refactor this method, based on our conversation.

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -1,4 +1,5 @@
-﻿using System;
+﻿using ConsoleTables;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
```
please remove the unused imports.

Alatrash814 commented on Inventory Management System/packages.config:
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="ConsoleTables" version="2.4.2" targetFramework="net472" />
```
Good Job.

Commit:
Refactor ProductClass to comply with style guidelines

In this commit, we made various changes to the ProductClass to ensure it adheres to the style guidelines. The following updates were implemented:

1. Removed extra spaces for a cleaner code layout.
2. Converted methods within the ProductClass to non-static methods, promoting better object-oriented design.
3. Improved code readability by organizing the switch statement into a separate method and handling invalid input gracefully.

These changes were applied exclusively to the ProductClass, resulting in a more maintainable and readable codebase, aligned with the specified style guidelines.

eid-sonkrot committed:
Refactor code style

eid-sonkrot committed:
Refactor code style

eid-sonkrot committed:
Merge remote-tracking branch 'origin/master' into productClass

eid-sonkrot committed:
change varible type in product class to var
